src/config/index.ts

```diff
@@ -195,22 +203,45 @@ export class Config {
   }
 
   public static getUniqueProperties(contactTypeName: string): ContactProperty[] {
+    const { config } = ConfigFactory.getConfigFactory().loadConfig(CONFIG_NAME);
     const contactMatch = config.contact_types.find(c => c.name === contactTypeName);
     const uniqueProperties = contactMatch?.place_properties.filter(prop => prop.unique);
     return uniqueProperties || [];
   }
 
+  public static isDefaultConfig(): boolean {
+    const { config } = ConfigFactory.getConfigFactory().loadConfig(CONFIG_NAME);
+    return config.domains.some(c => c.domain === 'default-configuration-domain');
+  }
+
   // TODO: Joi? Chai?
-  public static assertValid({ config }: PartnerConfig = partnerConfig) {
-    for (const contactType of config.contact_types) {
+  public static assertValid(config?: PartnerConfig) {
+    const { config: assertionConfig } = config || ConfigFactory.getConfigFactory().loadConfig(CONFIG_NAME);
+
+    if (!assertionConfig.domains || assertionConfig.domains.length === 0) {
+      throw Error(`invalid configuration: 'domains' property is empty`);
+    }
+
+    const availableDomains = Config.getDomains();
+    const newConfigHasExistingDomain = _.some(availableDomains,
+      existingDomain => _.some(assertionConfig.domains, newDomain => _.isMatch(existingDomain, newDomain)));
+    if (!this.isDefaultConfig() && !newConfigHasExistingDomain) {
+      throw Error(`Invalid configuration: 'domains' property doesn't contains any existing domain`);
+    }
+
+    if (!assertionConfig.contact_types || assertionConfig.contact_types.length === 0) {
+      throw Error(`invalid configuration: 'contact_types' property is empty`);
+    }
+
+    for (const contactType of assertionConfig.contact_types) {
```

Comment:
We need to update this and make validation very thorough making sure all required fields on the `ContactType` object have values. Use the object definition as a guide on what is strictly required. For eg. I can upload a `ContactType` without the `name` field even though it's required

Reply:
mainly focused on things that would crash the app

Reply:
Any required fields missing from `ContactType` will crash the app at some point. The app functionality i.e all generated forms, heavily depends on `ContactType`.

Reply:
i have tested this the app will only misbehave but wont crush unless there is a particular scenario im missing
if the app is misbehaving config can be update ideally it should be tested before deploying
at least for now 
lets handle these other cases in a new pr trying to get this in asap its blocking work

Reply:
The app won't work correctly if a `ContactType` doesn't have a valid `name`. The data clean-up you'd set yourself up for just from this one field missing is worth having the checks in place now. This is just one example.

Reply:
i dont think cht would accept uploading data if the `name` is invalid/wrong
and for this is unlikely to happen given config has to go thru testing by the dev before deployment
again this wont break the app lets handle all these scenarios in another pr